From: =?utf-8?b?TcOpc3rDoXJvcyBNaWjDoWx5?= <misi@niif.hu>
Date: Thu, 14 Dec 2017 13:01:56 +0100
Subject: empty cli-password not allowed=>disable telnet cli

Adds a warning to set a cli-password and disable telnet cli interface.
---
 src/apps/relay/mainrelay.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/apps/relay/mainrelay.c b/src/apps/relay/mainrelay.c
index 7f24c8c..2e57217 100644
--- a/src/apps/relay/mainrelay.c
+++ b/src/apps/relay/mainrelay.c
@@ -1977,6 +1977,11 @@ int main(int argc, char **argv)
 		exit(-1);
 	}
 
+	if(cli_password[0]==0) {
+		TURN_LOG_FUNC(TURN_LOG_LEVEL_ERROR, "\nCONFIG ERROR: Empty cli-password, and so telnet cli interface is disabled! Please set a non empty cli-password!\n");
+		use_cli = 0;
+	}
+
 	if(!use_lt_credentials && !anon_credentials) {
 		if(turn_params.default_users_db.ram_db.users_number) {
 			TURN_LOG_FUNC(TURN_LOG_LEVEL_WARNING, "\nCONFIGURATION ALERT: you specified long-term user accounts, (-u option) \n	but you did not specify the long-term credentials option\n	(-a or --lt-cred-mech option).\n 	I am turning --lt-cred-mech ON for you, but double-check your configuration.\n");
